src/spid-callbacks.js

```diff
@@ -0,0 +1,32 @@
+/*global module:false*/
+var _callbacks = {};
+
+function ensureArray(key) {
+    if(!_callbacks[key]) {
+        _callbacks[key] = [];
+    }
+}
+
+function register(key, cb) {
+    ensureArray(key);
+    _callbacks[key].push(cb);
+}
+
+function invokeAll(key, err, data) {
+    ensureArray(key);
+    _callbacks[key].forEach(function (cb) {
+        cb(err, data);
```

Comment:
if one `cb` throws, will it break the `forEach()` loop. https://stackoverflow.com/questions/2641347/how-to-short-circuit-array-foreach-like-calling-break

```javascript
try { cb(err, data); } catch (cbError) { logger.error(cbError); }
```

Reply:
Actually, all client given callbacks are wrapped in a makeAsync, and these just call the async version of the callback. If one throws, execution will go on as before.